src/lexer.js

```diff
@@ -15,6 +15,9 @@ class Lexer {
     }
 
     isIdentifier (id) {
+        /**
+         * @todo replace this with a dictionary matching, rather than use regex (much faster)
```

Comment:
Objects' key-value lookup have an [armotized time complexity of O(1)](https://www.quora.com/What-is-the-time-complexity-of-checking-if-a-key-is-in-a-dictionary-in-Python).

Can this regex exhibit similar?

Reply:
Yes it can. Classic regexes can be compiled into Deterministic Finite Automata that can match a string of length N in O(N) time. Our N in this case is the length of ```id``` and that will always be a single character. So we end up having O(1).

Reply:
Okay, then

Reply:
Alright then, so you can also remove the todo